### PR TITLE
[14.0][FIX] account_payment_purchase: solved error get payment mode and ban…

### LINKED ISSUE #899: [14.0] account_payment_purchase: changing many2one resets payment_mode
Hi,
We've seen that when the field purchase_vendor_bill_id is changed and a purchase is selected from it, the payment_mode_id is always reseted because it is using the reference purchase_id.
`new_mode = self.purchase_id.payment_mode_id.id or False`
We've made this change, and it seems to work as it should.
`new_mode = self.purchase_vendor_bill_id.purchase_order_id.payment_mode_id.id or False`
The same goes for the partner_bank_id field.
@MiquelRForgeFlow

### PR DESCRIPTION
**Pull request by cramosh**
…k id close #899

### CODE REVIEW

**Review by ae-landoo — APPROVED** [review]
Technical approval.

**Comment by Mat-moran** [comment]
thanks @ae-landoo we will try to pass the tests with ✅  as soon as we have a little bit of time.

**Comment by Mat-moran** [comment]
@ae-landoo all checks are OK, do you need to approve the PR or if everything is green the merge is automatic?

**Review by rousseldenis — APPROVED** [review]
Code review.

Maybe this deserves a little test?

**Comment by ae-landoo** [comment]
@HaraldPanten @pedrobaeza
Can it be merged?

**Comment by pedrobaeza** [comment]
Please fw-port it to v15:

/ocabot merge patch

**Comment by MiquelRForgeFlow** [comment]
The three commits should have been squashed before merge.

**Comment by pedrobaeza** [comment]
Oh, I didn't see it. Aborted. Please squash commits into one.

**Comment by cramosh** [comment]
Ok I squash the commits and upload it again

**Comment by cramosh** [comment]
Done

File: account_payment_purchase/models/account_invoice.py
Comment (willer-mo) [inline account_payment_purchase/models/account_invoice.py:21]: Little question, shoudn't this have into account that `purchase_vendor_bill_id` can also have a relation to an invoice?
So it would look something like this:
```suggestion
        new_mode = (
            self.purchase_vendor_bill_id.purchase_order_id.payment_mode_id.id
            or self.purchase_vendor_bill_id.vendor_bill_id.payment_mode_id.id
            or self.purchase_id.payment_mode_id.id
        )
        new_bank = (
            self.purchase_vendor_bill_id.purchase_order_id.supplier_partner_bank_id.id
            or self.purchase_vendor_bill_id.vendor_bill_id.partner_bank_id.id
            or self.purchase_id.supplier_partner_bank_id.id
        )
```

Comment (willer-mo) [reply account_payment_purchase/models/account_invoice.py:21]: Or maybe, add this on top of the function :thinking: :
```
if self.purchase_vendor_bill_id.vendor_bill_id:
    return super()._onchange_purchase_auto_complete()
```

**Pull request merged**
